tuning/sft_trainer.py

```diff
@@ -100,6 +98,9 @@ def train(
         Union[peft_config.LoraConfig, peft_config.PromptTuningConfig]
     ] = None,
     trainer_controller_args: configs.TrainerControllerArguments = None,
+    callbacks: Optional[List[TrainerCallback]] = None,
```

Comment:
```suggestion
    callbacks: Optional[List[TrainerCallback]] = [FileLoggingCallback()],
```

curious if we want to do this instead. While default is set in main(), please be aware not all adopting scripts will call from main() or call using command line. Main function is an example on how to call train.

Example : in our docker image that we ship to SW we call train() directly and dont use command line. https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/build/launch_training.py#L70

In that case if callback not set, do we want to set it to default FileLoggingCallback ? also is that logger arg in __init__ of FileLoggingCallback needed?

Reply:
@Ssukriti point taken but with this new API do you think that FileLoggingCallback should just be classified as another tracker which tracks logs to a file system. 
In that case we can set the callbacks default to a FileLoggingTracker.

On the logger arg in init of FileLoggingCallback, if I remove it the linter on my system complains of a `no-value-for-parameter` in that line.

```
tuning/sft_trainer.py:349:27: E1120: No value for argument 'logger' in constructor call (no-value-for-parameter)
```

Maybe we could make the logger argument optional? or altogether remove it as its not being called in the callback.

Reply:
I see you have now handled it in code by setting default callback in train() . What is done is fine and solves the purpose. You can alternatively create a FileLoggingTracker too and pass that as default, but to keep things simple whats done now also works

Reply:
you can mark this conversation as resolved if you want to stick to existing implementation as it solves thw purpose. Thanks

Reply:
looks good to me! Great change!